pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -200,6 +200,20 @@ public TopicPolicies getTopicPolicies(TopicName topicName,
                 : policiesCache.get(TopicName.get(topicName.getPartitionedTopicName()));
     }
 
+    @Override
+    public CompletableFuture<Void> removeTopicPoliciesCache(TopicName topicName) {
+        NamespaceName namespace = topicName.getNamespaceObject();
+        CompletableFuture<SystemTopicClient.Reader<PulsarEvent>> readerCompletableFuture =
```

Comment:
@Technoboy-  Hi, Just closing the reader doesn't seem to work. 

Have some producer will still be reconnection(Then the trigger topic is auto created again), May be also  should closed all writer?

And can only do it closed the readers/writers on the current broker. 

![Pasted Graphic](https://user-images.githubusercontent.com/33416836/147308436-1b6ca919-e59d-4594-be4d-4e02ac06f26a.png)

Essential reason, When close Pulsar client, Broker send client `CLOSE_PRODUCER `,  When the producer client receives the `CLOSE_PRODUCER`, it will reconnecting, Then the trigger topic is auto created again.

![Pasted Graphic 1](https://user-images.githubusercontent.com/33416836/147308429-7cab9d02-a9e7-4507-aa84-604555000f63.png)

Should we add new commands?  When the client receives the command, it only closes without  reconnecting.

Reply:
Get it. hard to get the writers reference, let me think how to process this case. Do you have any good idea?


> When the client receives the command, it only closes without reconnecting.

We may create a new PIP for discussing it.

Reply:
ok, I created PIP #13488.